python_modules/dagster/dagster/_daemon/run_coordinator/queued_run_coordinator_daemon.py

```diff
@@ -78,11 +78,17 @@ def run_iteration(
         workspace_process_context: IWorkspaceProcessContext,
         fixed_iteration_time: Optional[float] = None,  # used for tests
     ) -> DaemonIterator:
-        run_coordinator = workspace_process_context.instance.run_coordinator
-        run_queue_config = workspace_process_context.instance.get_run_queue_config()
-        if not isinstance(run_coordinator, QueuedRunCoordinator):
-            check.failed(f"Expected QueuedRunCoordinator, got {run_coordinator}")
-
+        if not isinstance(workspace_process_context.instance.run_coordinator, QueuedRunCoordinator):
+            check.failed(
+                f"Expected QueuedRunCoordinator, got {workspace_process_context.instance.run_coordinator}"
+            )
+        run_coordinator = cast(
+            QueuedRunCoordinator, workspace_process_context.instance.run_coordinator
+        )
```

Comment:
nit: is this fundamentally different from

`run_coordinator = check.inst(workspace_process_context.instance.run_coordinator, QueuedRunCoordinator)`?